app/src/components/common/button_disconnect_wallet/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { useWeb3Context } from 'web3-react/dist'
+
+const Separator = css`
+  &::before {
+    background-color: ${props => props.theme.header.color};
+    content: '';
+    height: 22px;
+    margin: 0 15px;
+    width: 1px;
+  }
+`
+
+const LogoutWrapper = styled.div`
+  align-items: center;
+  display: flex;
+  ${Separator}
+`
+
+const ButtonDisconnectWalletStyled = styled.div`
+  width: 166px;
+  height: 28px;
+  border-radius: 2px;
+  background-color: #ff7848;
+  cursor: pointer;
+`
+
+const ButtonTitle = styled.div`
+  width: 144px;
+  height: 18px;
+  font-family: Roboto;
+  font-size: 13px;
+  font-weight: 500;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: 1.38;
+  letter-spacing: normal;
+  text-align: center;
+  color: #ffffff;
+  margin-top: 5.4px;
+  margin-left: 11px;
+`
+
+interface Props {
+  callback: () => void
+}
+
+export const ButtonDisconnectWallet = (props: Props) => {
+  const context = useWeb3Context()
```

Comment:
You could use `useConnectedWeb3Context` here and avoid all the unnecessary checks below. The problem is that, I think, you won't have the `unsetConnector` callback. I wonder if it would make sense to expose the "raw" web3 context inside the connected web3 context (an extra `web3Context` property maybe?).

Reply:
Yeahh, thats why I use this context, because I need to use the unsetConnector, mmm I will try to expose the context